src/components/flyout/flyout.styles.ts

```diff
@@ -0,0 +1,257 @@
+      flex-direction: column;
+      align-items: stretch;
+      &:focus {
+        outline: none;
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('max-width', '90vw')}
+      }
+    `,
+
+    // Flyout Sizes
+    s: css`
+      ${composeFlyoutSizing(euiThemeContext, 's')}
+    `,
+    m: css`
+      ${composeFlyoutSizing(euiThemeContext, 'm')}
+    `,
+    l: css`
+      ${composeFlyoutSizing(euiThemeContext, 'l')}
+    `,
+
+    // Side
+    right: css`
+      clip-path: polygon(-50% 0, 100% 0, 100% 100%, -50% 100%);
+      ${logicalCSS('border-left', euiTheme.border.thin)}
+      ${logicalCSS('right', 0)}
+      ${euiCanAnimate} {
+        animation: ${euiFlyout} ${euiTheme.animation.normal}
+          ${euiTheme.animation.resistance};
+      }
+    `,
+    left: css`
+      ${logicalCSS('border-right', euiTheme.border.thin)}
+      ${logicalCSS('left', 0)}
+      clip-path: polygon(0 0, 150% 0, 150% 100%, 0 100%);
+      ${euiCanAnimate} {
+        animation: ${euiFlyoutLeft};
+      }
+    `,
+
+    // Type
+    overlay: css``,
+    push: css`
+      box-shadow: none;
+      clip-path: none;
+      animation-duration: 0s !important; // Don't animate on loading a docked nav
+      // Make sure the header shadows are above
+      z-index: ${Number(euiTheme.levels.flyout) - 1};
+    `,
+    pushSide: {
+      right: css`
+        ${logicalCSS('border-left', euiTheme.border.thick)}
+      `,
+      left: css`
+        ${logicalCSS('border-right', euiTheme.border.thick)}
+      `,
+    },
+
+    // Padding
+    paddingSizes: {
+      none: css`
+        ${composeFlyoutPadding(euiThemeContext, 'none')}
+      `,
+      s: css`
+        ${composeFlyoutPadding(euiThemeContext, 's')}
+      `,
+      m: css`
+        ${composeFlyoutPadding(euiThemeContext, 'm')}
+      `,
+      l: css`
+        ${composeFlyoutPadding(euiThemeContext, 'l')}
+      `,
+    },
+  };
+};
+
+const composeFlyoutSizing = (
+  euiThemeContext: UseEuiTheme,
+  size: EuiFlyoutSize
+) => {
+  const euiTheme = euiThemeContext.euiTheme;
+
+  const flyoutSizes = {
+    s: {
+      min: `${Math.round(euiTheme.breakpoint.m * 0.5)}px`,
+      width: '25vw',
+      max: `${Math.round(euiTheme.breakpoint.s * 0.7)}px`,
+    },
+
+    m: {
+      min: `${euiTheme.breakpoint.m * 0.8}px`,
+      width: '50vw',
+      max: euiTheme.breakpoint.m,
+    },
+
+    l: {
+      min: `${Math.round(euiTheme.breakpoint.m * 0.9)}px`,
+      width: '75vw',
+      max: `${Math.round(euiTheme.breakpoint.l)}px`,
+    },
+  };
+
+  const flyoutSizing = css`
+    ${logicalCSS('max-width', flyoutSizes[size].max)}
+
+    ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+      ${logicalCSS('min-width', flyoutSizes[size].min)}
+      ${logicalCSS('width', flyoutSizes[size].width)}
+    }
+    ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+      ${logicalCSS('min-width', 0)}
+      ${logicalCSS('width', flyoutSizes[size].min)}
+    }
+  `;
+
+  return flyoutSizing;
```

Comment:
ah, just noticed this when looking at the snapshots - when you nest a css literal within a css, Emotion ends up concatenating the  classNames in a weird way, so just want a template literal `` here.

I also think we can drop the var declaration since it's returned immedately
```suggestion
  return `
    ${logicalCSS('max-width', flyoutSizes[size].max)}

    ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
      ${logicalCSS('min-width', flyoutSizes[size].min)}
      ${logicalCSS('width', flyoutSizes[size].width)}
    }
    ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
      ${logicalCSS('min-width', 0)}
      ${logicalCSS('width', flyoutSizes[size].min)}
    }
  `;
```